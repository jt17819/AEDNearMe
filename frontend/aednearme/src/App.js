import React from "react";
import { Route, Routes, Outlet } from 'react-router-dom';
import Home from "./pages/Home/Home";
import AEDhowto from "./pages/AED/AEDhowto";
import CPRhowto from "./pages/CPR/CPRhowto";
import Quiz from "./pages/Quiz/Quiz";
import Choking from "./pages/Choking/Choking";

import Login from "./pages/Login/Login";
import Register from "./pages/Register/Register";
import Upload from "./pages/Upload/Upload";
import FAQ from "./pages/FAQ/FAQ";

import { NavBar } from "./components/NavBar/NavBar";
import { Footer } from "./components/Footer/Footer.js";

import './App.css';

function App() {
  return (
    <div className="App">
      <NavBar />
      <Routes>
          <Route path='/' element={<Home />} />
          <Route path='/aedhowto' element={<AEDhowto />} />
          <Route path='/cprhowto' element={<CPRhowto />} />
          <Route path='/quiz' element={<Quiz />} />
          <Route path='/choking' element={<Choking />} />
          <Route path='/login' element={<Login />} />
          <Route path='/register' element={<Register />} />
          <Route path='/upload' element={<Upload />} />
          <Route path='/faq' element={<FAQ />} />
   
          <Route path='*' element={<p>nothing to see here, move on mate.</p>} />
      </Routes>

      <Footer />

      <Outlet />
    </div>
  );
}

export default App;
