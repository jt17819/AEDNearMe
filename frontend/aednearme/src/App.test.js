import { render, fireEvent, screen, cleanup, within } from '@testing-library/react';
import { MemoryRouter, Router as Router } from 'react-router-dom';
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history'
import App from './App';
import axios from 'axios'


describe('App', () => {

  // beforeEach(() => {
  //   render(<App />, { wrapper: MemoryRouter })
  // });



  // Rendering

  it('renders welcome message', () => {
    render(<App />, { wrapper: MemoryRouter })
    expect(screen.getByText('IN A LIFE THREATENING EMERGENCY, CALL 999 or 112')).toBeInTheDocument();
  });

  test('list contains 2 buttons', () => {
    render(<App />, { wrapper: MemoryRouter });
  
    // const listElement = screen.getAllByRole('button');
    const listItems = screen.getAllByRole('button');
  
    
    expect(listItems.length).toEqual(2);
  });



  // Snapshot

  it('should take a snapshot', () => {
    render(<App />, { wrapper: MemoryRouter })
    const { asFragment } = render(<App />, { wrapper: MemoryRouter })
    
    expect(asFragment(<App />)).toMatchSnapshot()
  })

  

  
})

