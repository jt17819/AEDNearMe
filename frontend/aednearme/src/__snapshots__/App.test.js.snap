// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appbar-34 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          class="makeStyles-root-29"
        >
          <div
            class=""
          >
            <a
              class="makeStyles-logo-31"
              href="/"
            >
              AED 
              <br />
               nearMe 
            </a>
          </div>
          <div
            class="makeStyles-navlinks-30"
          >
            <a
              class="makeStyles-link-32"
              href="/"
            >
              Home
            </a>
            <div>
              <span
                aria-haspopup="true"
                data-testid="span"
                id="basic-button"
                style="color: rgb(255, 255, 255); font-size: 20px; cursor: pointer; margin-left: 20px; font-weight: bold;"
              >
                AED
              </span>
            </div>
            <div>
              <span
                aria-haspopup="true"
                data-testid="span"
                id="basic-button"
                style="color: rgb(255, 255, 255); font-size: 20px; cursor: pointer; margin-left: 20px; font-weight: bold;"
              >
                CPR
              </span>
            </div>
            <div>
              <span
                aria-haspopup="true"
                data-testid="span"
                id="basic-button"
                style="color: rgb(255, 255, 255); font-size: 20px; cursor: pointer; margin-left: 20px; font-weight: bold;"
              >
                Choking
              </span>
            </div>
            <div>
              <span
                aria-haspopup="true"
                data-testid="span"
                id="basic-button"
                style="color: rgb(255, 255, 255); font-size: 20px; cursor: pointer; margin-left: 20px; font-weight: bold;"
              >
                Education Hub
              </span>
            </div>
            <a
              class="makeStyles-link-32"
              href="/login"
            >
              Login
            </a>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root pb-5 MuiContainer-maxWidthLg"
      style="padding-bottom: 100px; padding-top: 20px;"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-5"
        >
          <h1
            class="makeStyles-title-35"
          >
            IN A LIFE THREATENING EMERGENCY, CALL 999 or 112
          </h1>
          <p
            class="makeStyles-text-36"
          >
            Use our location driven database to show your nearest AED.
          </p>
          <p
            class="makeStyles-text-36"
          >
            An AED machine, or Automated External Defibrillator, is an essential component in first aid and safety.
          </p>
          <p
            class="makeStyles-text-36"
          >
            Most of us have seen these units at airports, hospitals, schools, government buildings, nursing homes, first aid stations and even in our workplaces â€” But do we remember where we seen them ?
          </p>
          <div
            class="makeStyles-btn_box-37"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-primary-38"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                How to use AED
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-secondary-39"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Add new Location
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7"
        >
          <div>
            <div />
            <div>
              Loading...
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="makeStyles-footer-40"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-between"
          >
            <div
              class="MuiGrid-root makeStyles-item-41 MuiGrid-item"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge makeStyles-snsIcon-42 css-tzssek-MuiSvgIcon-root"
                data-testid="MarkEmailUnreadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 8.98V18c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2h10.1c-.06.32-.1.66-.1 1 0 1.48.65 2.79 1.67 3.71L12 11 4 6v2l8 5 5.3-3.32c.54.2 1.1.32 1.7.32 1.13 0 2.16-.39 3-1.02zM16 5c0 1.66 1.34 3 3 3s3-1.34 3-3-1.34-3-3-3-3 1.34-3 3z"
                />
              </svg>
              <p>
                eMail us if any query: 
                <br />
                 info@AEDnearMe.com 
              </p>
            </div>
            <div
              class="MuiGrid-root makeStyles-item-41 MuiGrid-item"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-snsIcon-42 css-i4bv87-MuiSvgIcon-root"
                data-testid="CopyrightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.88 9.14c1.28.06 1.61 1.15 1.63 1.66h1.79c-.08-1.98-1.49-3.19-3.45-3.19C9.64 7.61 8 9 8 12.14c0 1.94.93 4.24 3.84 4.24 2.22 0 3.41-1.65 3.44-2.95h-1.79c-.03.59-.45 1.38-1.63 1.44-1.31-.04-1.86-1.06-1.86-2.73 0-2.89 1.28-2.98 1.88-3zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
              <p>
                2022 
                <br />
                 by Snake in the Flask 
              </p>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
